staticaddr/loopin/manager.go

```diff
@@ -189,12 +206,217 @@ func (m *Manager) Run(ctx context.Context, currentHeight uint32) error {
+			err = m.handleLoopInSweepReq(ctx, sweepReq)
+			if err != nil {
+				log.Errorf("Error handling loop-in sweep request: %v",
+					err)
+			}
+
 		case <-ctx.Done():
 			return ctx.Err()
 		}
 	}
 }
 
+// notifyNotFinished notifies the server that a swap is not finished by
+// sending an empty signature map.
+func (m *Manager) notifyNotFinished(ctx context.Context, swapHash lntypes.Hash,
+	txId chainhash.Hash) error {
+
+	_, err := m.cfg.Server.PushStaticAddressSweeplessSigs(
+		ctx, &looprpc.PushStaticAddressSweeplessSigsRequest{
+			SwapHash: swapHash[:],
+			Txid:     txId[:],
+		})
+
+	return err
+}
+
+// handleLoopInSweepReq handles a loop-in sweep request from the server.
+// It first checks if the requested loop-in is finished as expected and if
+// yes will send signature to the server for the provided psbt.
+func (m *Manager) handleLoopInSweepReq(ctx context.Context,
+	req *swapserverrpc.ServerStaticLoopInSweepNotification) error {
+
+	// First we'll check if the loop-ins are known to us and in
+	// the expected state.
+	swapHash, err := lntypes.MakeHash(req.SwapHash)
+	if err != nil {
+		return err
+	}
+
+	// Fetch the loop-in from the store.
+	loopIn, err := m.cfg.Store.FetchLoopInByHash(ctx, swapHash)
+	if err != nil {
+		return err
+	}
+
+	loopIn.AddressParams, err =
+		m.cfg.AddressManager.GetStaticAddressParameters(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	loopIn.Address, err = m.cfg.AddressManager.GetStaticAddress(
+		ctx,
+	)
+	if err != nil {
+		return err
+	}
+
+	reader := bytes.NewReader(req.SweepTxPsbt)
+	sweepPacket, err := psbt.NewFromRawBytes(reader, false)
+	if err != nil {
+		return err
+	}
+
+	sweepTx := sweepPacket.UnsignedTx
+
+	// If the loop-in is not in the Succeeded state we return an
+	// error.
+	if loopIn.state != Succeeded {
+		// We'll notify the server that we don't consider the swap
+		// finished yet, so it can retry later.
+		_ = m.notifyNotFinished(ctx, swapHash, sweepTx.TxHash())
+		return fmt.Errorf("loop-in %v not in Succeeded state",
+			swapHash)
+	}
+
+	// Perform a sanity check on the number of unsigned tx inputs and
+	// prevout info.
+	if len(sweepTx.TxIn) != len(req.PrevoutInfo) {
+		return fmt.Errorf("expected %v inputs, got %v",
+			len(req.PrevoutInfo), len(sweepTx.TxIn))
+	}
+
+	// Check if all the deposits requested are part of the loop-in and
+	// find them in the requested sweep.
+	depositToIdxMap := make(map[string]int)
+	for reqOutpoint := range req.DepositToNonces {
+		hasDeposit := false
+		for _, depositOutpoint := range loopIn.DepositOutpoints {
+			if depositOutpoint == reqOutpoint {
+				hasDeposit = true
+				break
+			}
+		}
+		if !hasDeposit {
+			return fmt.Errorf("deposit outpoint not part of " +
+				"loop-in")
+		}
+
+		foundDepositInTx := false
+
+		for i, txIn := range sweepTx.TxIn {
+			if txIn.PreviousOutPoint.String() == reqOutpoint {
+				depositToIdxMap[reqOutpoint] = i
+				foundDepositInTx = true
+				break
+			}
+		}
+
+		if !foundDepositInTx {
```

Comment:
Do we need this check? IIUC it prevents sweeping a subset of deposits. What if we need to do this? E.g. because the number of inputs in a batch transaction is too high and we can accept some items to an existing batch, but not all. I think, current sweepbatcher logic can do this.